front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -40,21 +42,31 @@ const useScenarioData = (
 
   const projectionPath = usePathProjection(infra);
 
-  const {
-    trainScheduleSummariesById,
-    projectedTrainsById,
-    setTrainScheduleSummariesById,
-    setProjectedTrainsById,
-    allTrainsProjected,
-    allTrainsLoaded,
-  } = useLazyLoadTrains({
+  const { trainScheduleSummariesById, setTrainScheduleSummariesById, allTrainsLoaded } =
+    useLazyLoadTrains({
+      infraId: scenario.infra_id,
+      trainIdsToFetch,
+      setTrainIdsToFetch,
+      trainSchedules,
+      setTrainIdsToProject,
+    });
+
+  const { projectedTrainsById, allTrainsProjected, setProjectedTrainsById } = useLazyProjectTrains({
     infraId: scenario.infra_id,
-    trainIdsToFetch,
-    setTrainIdsToFetch,
+    trainIdsToProject,
     path: projectionPath?.path,
     trainSchedules,
+    moreTrainsToCome: !allTrainsLoaded,
+    setTrainIdsToProject,
   });
 
+  useEffect(() => {
+    if (trainSchedules && projectionPath?.path) {
```

Comment:
Previously, there was a `allTrainsLoaded` check here as well. Has it been removed on purpose?

Reply:
WFM!